src/lib/y2network/config_reader/sysconfig_routes_reader.rb

```diff
@@ -0,0 +1,122 @@
+      def config
+        routes = load_routes.map { |r| build_route(r) }
+        Y2Network::RoutingTable.new(routes)
+      end
+
+    private
+
+      MISSING_VALUE = "-".freeze
+      private_constant :MISSING_VALUE
+      DEFAULT_DEST = "default".freeze
+      private_constant :DEFAULT_DEST
+
+      # Loads routes from system
+      #
+      # @return [Array<Hash<String, String>>] list of hashes representing routes
+      #                                       as provided by SCR agent.
+      #                                       keys: destination, gateway, netmask, [device, [extrapara]]
+      def load_routes
+        routes = Yast::SCR.Read(@routes_file) || []
+        normalize_routes(routes.uniq)
+      end
+
+      # Converts routes config as read from system into well-defined format
+      #
+      # Expects list of hashes as param. Hash should contain keys "destination",
+      # "gateway", "netmask", "device", "extrapara"
+      #
+      # Currently it converts "destination" in CIDR format (<ip>/<prefix_len>)
+      # and keeps just <ip> part in "destination" and puts "/<prefix_len>" into
+      # "netmask"
+      #
+      # @param routes [Array<Hash>] in quad or CIDR flavors (see {#Routes})
+      # @return [Array<Hash>] in quad or slash flavor
+      def normalize_routes(routes)
+        return routes if routes.nil? || routes.empty?
+
+        routes.map do |route|
+          subnet, prefix = route["destination"].split("/")
+
+          next route if prefix.nil?
+
+          route["destination"] = subnet
+          route["netmask"] = "/#{prefix}"
+
+          route
+        end
+      end
+
+      # Given an IP and a netmask, returns a valid IPAddr object
+      #
+      # @param ip_str      [String] IP address; {MISSING_VALUE} means that the IP is not defined
+      # @param netmask_str [String] Netmask; {MISSING_VALUE} means than no netmask was specified
+      # @return [IPAddr,nil] The IP address or `nil` if the IP is missing
+      def build_ip(ip_str, netmask_str = MISSING_VALUE)
+        return nil if ip_str == MISSING_VALUE
+
+        ip = IPAddr.new(ip_str)
+        netmask_str == MISSING_VALUE ? ip : ip.mask(netmask_str)
+      end
+
+      # Build a route given a hash from the SCR agent
+      #
+      # @param hash [Hash] Hash from the `.routes` SCR agent
+      # @return Route
+      def build_route(hash)
+        # TODO: check whether the iface is configured in the system
+        iface = Interface.new(hash["device"])
+        # normalized SCR output contains either subnet mask or /<prefix length> under
+        # "netmask" key
+        # TODO: this should be improved in normalize_routes
+        mask = hash["netmask"] =~ /\/[0-9]+/ ? hash["netmask"][1..-1] : hash["netmask"]
+
+        Y2Network::Route.new(
+          to:        hash["destination"] != DEFAULT_DEST ? build_ip(hash["destination"], mask) : :default,
+          interface: iface,
+          gateway:   build_ip(hash["gateway"], MISSING_VALUE)
```

Comment:
options is not imported

Reply:
will fix when merging the ifroute PR (need to fix its agent which ignores options anyway)